msu/systems/serialization/serde_emulator.nut

```diff
@@ -1,45 +1,170 @@
 // Base for the Serialization and Deserialization Emulators
 ::MSU.Class.SerDeEmulator <- class
 {
-	static __IDRegex = regexp("^.*\\.\\d+$")
-	Data = null;
-	Mod = null;
-	ID = null;
 	MetaData = null;
-	FlagContainer = null;
+	SerializationData = null;
+	Idx = -1;
```

Comment:
This should be moved into the deserialization emulators only, so into `__ReadFunctions`, which means that it should probably be renamed into `__ReadFields`